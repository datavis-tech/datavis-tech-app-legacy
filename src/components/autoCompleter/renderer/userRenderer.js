import { Label } from 'semantic-ui-react'

export default ({ title }) => <Label content={title} />
