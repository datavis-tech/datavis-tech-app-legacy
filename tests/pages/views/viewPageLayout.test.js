import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import nodeSelector from '../../utils/nodeSelector'
import ViewPageLayout from '../../../src/pages/views/viewPageLayout'

describe('view page layout', () => {

  let sut
  let props
  let Header
  let Content
  let Description
  let References
  let ForkedFrom
  let Avatar
  let EditButton
  let ForkButton

  beforeEach(() => {
    Header = <div data-test='Header' />
    Content = <div data-test='Content' />
    Description = <div data-test='Description' />
    References = <div data-test='References' />
    ForkedFrom = <div data-test='ForkedFrom' />
    Avatar = <div data-test='Avatar' />
    EditButton = <div data-test='EditButton' />
    ForkButton = <div data-test='ForkButton' />

    props = {
      Header,
      Content,
      Description,
      References,
      ForkedFrom,
      Avatar,
      EditButton,
      ForkButton
    }

    sut = mount(<ViewPageLayout {...props} />)
  })

  it('should render header', () => {
    expect(sut.find(nodeSelector('Header')).exists()).toBeTruthy()
  })

  it('should render content', () => {
    expect(sut.find(nodeSelector('Content')).exists()).toBeTruthy()
  })

  it('should render description', () => {
    expect(sut.find(nodeSelector('Description')).exists()).toBeTruthy()
  })

  it('should render references', () => {
    expect(sut.find(nodeSelector('References')).exists()).toBeTruthy()
  })

  it('should render forked from', () => {
    expect(sut.find(nodeSelector('ForkedFrom')).exists()).toBeTruthy()
  })

  it('should render avatar', () => {
    expect(sut.find(nodeSelector('Avatar')).exists()).toBeTruthy()
  })

  it('should render edit button', () => {
    expect(sut.find(nodeSelector('EditButton')).exists()).toBeTruthy()
  })

  it('should render fork button', () => {
    expect(sut.find(nodeSelector('ForkButton')).exists()).toBeTruthy()
  })

  it('should render correct layout', () => {
    const tree = renderer
      .create(<ViewPageLayout {...props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})
