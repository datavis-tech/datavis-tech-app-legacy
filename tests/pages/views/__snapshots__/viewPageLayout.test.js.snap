// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view page layout should render correct layout 1`] = `
<div>
  <div
    data-test="Header"
  />
  <div
    data-test="Content"
  />
  <div
    className="ui divided two column grid"
  >
    <div
      className="row"
    >
      <div
        className="twelve wide column"
      >
        <div
          data-test="Avatar"
        />
        <div
          data-test="Description"
        />
        <div
          data-test="References"
        />
        <div
          data-test="ForkedFrom"
        />
      </div>
      <div
        className="center aligned four wide column"
      >
        <div
          data-test="EditButton"
        />
        <div
          data-test="ForkButton"
        />
        <div
          data-test="EmbedButton"
        />
      </div>
    </div>
  </div>
</div>
`;
